astrix-integration-tests/src/test/java/com/avanza/astrix/integration/tests/TracingTests.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2014 Avanza Bank AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.avanza.astrix.integration.tests;
+
+import com.avanza.astrix.beans.core.AstrixSettings;
+import com.avanza.astrix.config.DynamicConfig;
+import com.avanza.astrix.config.GlobalConfigSourceRegistry;
+import com.avanza.astrix.config.MapConfigSource;
+import com.avanza.astrix.context.AstrixConfigurer;
+import com.avanza.astrix.context.AstrixContext;
+import com.avanza.astrix.gs.test.util.PuConfigurers;
+import com.avanza.astrix.gs.test.util.RunningPu;
+import com.avanza.astrix.integration.tests.domain.api.LunchService;
+import com.avanza.astrix.provider.component.AstrixServiceComponentNames;
+import com.avanza.astrix.test.util.AutoCloseableRule;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.Properties;
+
+import static com.avanza.astrix.integration.tests.TestLunchRestaurantBuilder.lunchRestaurant;
+
+public class TracingTests {
```

Comment:
Very nice! @oskarwiksten

Reply:
Credit and praise for this test goes to @joakimsahlstrom though :)